src/Chapter4.hs

```diff
@@ -489,6 +510,17 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+concatList :: List a -> List a -> List a
+concatList (Cons x xs) ys = Cons x (concatList xs ys)
```

Comment:
Perfect!
Usually in Haskell, functions that add a few structures into are called `append*`, and contact is usually used for nested structured in order to flat them, but that is okay 🙂 

You can one more case to make your solution even more efficient:

```haskell
append xs Empty = xs
```

In that case, you won't need to walk through each element of the first list when the second one is empty and return the list instantly 🚤